frontend/javascript/components/bs-modal.vue

```diff
@@ -0,0 +1,89 @@
+<script setup>
+import { ref, onMounted, nextTick } from 'vue'
+import { Modal } from 'bootstrap'
+
+const props = defineProps({
+  dialogClasses: {
+    type: String,
+    default: ''
+  },
+  contentClasses: {
+    type: String,
+    default: ''
+  },
+  showOnMounted: {
+    type: Boolean,
+    default: false
+  }
+})
+
+const modalEl = ref()
+let bsModal
+
+// Current behavior is "keep-alive"-ish:
+// modal markup will be rendered on first show,
+// then hidden, not unmounted.
+// Consider using a v-model here...
+const doRender = ref(false)
+
+const show = async () => {
+  if (!doRender.value) {
+    doRender.value = true
+    await nextTick()
+    initialize()
+  }
+  bsModal.show()
+}
+
+const hide = () => {
+  bsModal.hide()
+}
+
+const initialize = ()  => {
+  bsModal = new Modal(modalEl.value)
+}
+
+onMounted(() => {
+  if (props.showOnMounted) {
+    show()
+  }
+})
+
+defineExpose({
+  show,
+  hide
+})
+</script>
+
+<template>
+  <Teleport to="body">
+    <div
+      ref="modalEl"
+      class="modal"
+      tabindex="-1"
+      role="dialog"
+      v-if="doRender"
+      >
+      <div
+        :class="'modal-dialog ' + dialogClasses"
```

Comment:
Instead, do this:


```suggestion
        v-bind="$attrs"
```

This will bind all (non-prop) attributes to the element. So not just classes work!

Reply:
I'd add the `v-bind` if it were the outermost `<div>`, the one with `class="modal"`, to allow `<bs-modal class="fade"`... there is also `contentClasses` right after it. yagni? too much bike shedding?